core/market/decentralized/tests/test_agreement.rs

```diff
@@ -1,45 +1,615 @@
-        .requestor_engine
-        .create_agreement(identity1.clone(), &proposal_id, Utc::now())
+    let agreement_id = req_engine
+        .create_agreement(req_id.clone(), &proposal_id, Utc::now())
         .await?;
-    let agreement = bus::service(network.node_gsb_prefixes(node_id1).0)
+    let agreement = bus::service(network.node_gsb_prefixes(REQ_NAME).0)
         .send(market::GetAgreement {
             agreement_id: agreement_id.to_string(),
         })
         .await??;
     assert_eq!(agreement.agreement_id, agreement_id.to_string());
     assert_eq!(
         agreement.demand.requestor_id.unwrap(),
-        identity1.identity.to_string()
+        req_id.identity.to_string()
     );
     assert_eq!(
         agreement.offer.provider_id.unwrap(),
-        identity2.identity.to_string()
+        prov_id.identity.to_string()
+    );
+    Ok(())
+}
+
+#[cfg_attr(not(feature = "market-test-suite"), ignore)]
+#[actix_rt::test]
+async fn full_market_interaction_aka_happy_path() -> Result<()> {
+    let network = MarketsNetwork::new("full_market_interaction_aka_happy_path")
+        .await
+        .add_market_instance(REQ_NAME)
+        .await?
+        .add_market_instance(PROV_NAME)
+        .await?;
+
+    let proposal_id = exchange_draft_proposals(&network, REQ_NAME, PROV_NAME).await?;
+    let req_market = network.get_market(REQ_NAME);
+    let req_engine = &req_market.requestor_engine;
+    let req_id = network.get_default_id(REQ_NAME);
+
+    // Requestor creates agreement with 1h expiration
+    let agreement_id = req_engine
+        .create_agreement(
+            req_id.clone(),
+            &proposal_id,
+            Utc::now() + Duration::hours(1),
+        )
+        .await?;
+
+    // Confirms it immediately
+    req_engine
+        .confirm_agreement(req_id.clone(), &agreement_id)
+        .await?;
+
+    // And starts waiting for Agreement approval by Provider
+    let agr_id = agreement_id.clone();
+    let query_handle = tokio::spawn(async move {
+        let approval_status = req_market
+            .requestor_engine
+            .wait_for_approval(&agr_id, 0.1)
+            .await?;
+
+        assert_eq!(
+            approval_status.to_string(),
+            ApprovalStatus::Approved.to_string()
+        );
+        Result::<(), anyhow::Error>::Ok(())
+    });
+
+    // Provider approves the Agreement and waits for ack
+    network
+        .get_market(PROV_NAME)
+        .provider_engine
+        .approve_agreement(network.get_default_id(PROV_NAME), &agreement_id, 0.1)
+        .await?;
+
+    // Protect from eternal waiting.
+    tokio::time::timeout(Duration::milliseconds(150).to_std()?, query_handle).await???;
+
+    Ok(())
+}
+
+// TODO: It is allowed in general, but probably after rejection or expiration??
+// TODO: but we don't know even how we should handle this case
+//#[cfg_attr(not(feature = "market-test-suite"), ignore)]
+#[ignore]
+#[actix_rt::test]
+async fn second_creation_should_fail() -> Result<()> {
```

Comment:
Is it possible to have a name conflict with test from other teams?

Reply:
nope, tests need to be unique just within single file or even module